### PR TITLE
Refactor DateSelect

### PR DESCRIPTION
**Pull request by ramchale**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | yes
| QA            | no

### Description

Refactor DateSelect to be consistent with new MonthSelect behaviour

### CODE REVIEW

File: docs/book/v3/standard-filters.md [hunk 1]
```diff
@@ -706,7 +729,7 @@ This will return `-4.4` (as a float).
 
 ## MonthSelect
 
-`Laminas\Filter\MonthSelect` allows you to filter a month and year value into a hyphen dash string.
+`Laminas\Filter\MonthSelect` allows you to filter a month and year value into a dash seperated string.
```
Comment (samsonasik) [inline docs/book/v3/standard-filters.md]: typo: seperated be separated?

Comment (ramchale) [reply docs/book/v3/standard-filters.md]: Thanks

**Review by froschdesign — APPROVED** [review]
Looks very good to me! 👍🏻 

@gsteel 
What is your opinion? Have I missed something?

**Review by gsteel — APPROVED** [review]
Cracking! Thanks @ramchale

**Pull request merged**